http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/HttpOrHttp2ChannelPool.java

```diff
@@ -171,7 +171,22 @@ private void release0(Channel channel, Promise<Void> promise) {
 
     @Override
     public void close() {
-        doInEventLoop(eventLoop, protocolImpl::close);
+        doInEventLoop(eventLoop, this::close0);
     }
 
+    private void close0() {
+        if (protocolImplPromise == null) {
+            return;
+        }
+
+        protocolImplPromise.addListener((Future<ChannelPool> f) -> {
+            if (f.isSuccess()) {
+                f.getNow().close();
```

Comment:
I wonder if we should wait for close to finish.

Reply:
I had the same though but https://netty.io/4.0/api/io/netty/channel/pool/ChannelPool.html#close--